tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/unaryoperations/UnaryExprTest.java

```diff
@@ -199,6 +199,11 @@ public void testNullableUnaryExpressions() {
         BRunUtil.invoke(result, "testNullableUnaryExpressions");
     }
 
+    @Test(description = "Test unary operators with user defined subtypes")
```

Comment:
Let's also add some negative tests for the static type of the unary expression.

e.g.,
```ballerina
type A -2|-1|0|1|2;
type B 0f|1f;

public function main() {
    A x = -2;
    A _ = -x; // error

    B y = 0f;
    B _ = -y; // error
}
```

And unary expressions where the expression is of a static type that contains multiple numeric types or a mix of numeric and non-numeric types.

e.g.,

```ballerina
type A 0|1f;
type B 0|1|"ABC";

public function main() {
    A x = 0;
    int _ = -x; // error

    B y = 0;
    int _ = -y; // error
}
```